download/download.go

```diff
@@ -103,7 +103,7 @@ func (d *Downloader) Start(ctx context.Context) {
 }
 
 func (d *Downloader) doStart(ctx context.Context) {
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
❓ Are you sure that this one wasn't intentionally overridden? The loop may well go on on running after the start is done.

I guess we could drop the `ctx` argument to `doStart`, if we don't use it to guard _starting taking too long_.

Reply:
I think it works as creating a child context. As I have it, this stops downloads if:

- the parent context passed to `Start()` is cancelled, or
- `Stop()` is called.

The case where we'd want to create a new context is if we want the download to continue *even if* the Context passed to `doStart()` is cancelled, effectively making the download only stoppable by calling `Stop()`. We actually can't exit `doStart()` until `go d.loop(ctx)` passes a value to the stop channel as we block on `done := <-d.stop`.

The only case where that happens in `Stop()` - nothing else writes to `d.stop`.

(Happy to revert either way)

Reply:
Yeah I'm torn about this one. Your argument makes sense,but introduces a change in behavior. I'd go for consistency now -- Let's do where what the other plugins do.... it seems some even pass no ctx at all to their loop construct.

How about we keep it as-is and reconsider this later? 💭

Reply:
Works for me! Reverted.